src/features/Sync/SyncOutManager.ts

```diff
@@ -0,0 +1,94 @@
+}
+
+const PATHS = {
+    LOGIN: 'login',
+    SENSOR: 'sensor', 
+    TEMPERATURE_LOG: 'temperature-log',
+    TEMPERATURE_BREACH: 'temperature-breach'
+}
+class SyncOutManager {
+    private paths: { [key: string]: string };
+
+    constructor() {
+        // TODO: initialise SyncOut manager from dependency container.
+        const rootPath = `http://${HOSTNAME}:${PORT}/${RESOURCES.COLDCHAIN_V1}`;
+
+        const loginPath = `${rootPath}/${PATHS.LOGIN}`;
+        const sensorPath = `${rootPath}/${PATHS.SENSOR}`;
+        const temperatureLogPath = `${rootPath}/${PATHS.TEMPERATURE_LOG}`;
+        const temperatureBreachPath = `${rootPath}/${PATHS.TEMPERATURE_BREACH}`;
+
+        this.paths = {
+            login: loginPath,
+            sensor: sensorPath,
+            temperatureLog: temperatureLogPath,
+            temperatureBreach: temperatureBreachPath,
+        };
+    }
+
+    
+
+    pushLogs = async(logs: SyncLog[]) => {
+        const loginResponse = await fetch(this.paths.login, {
+            method: 'POST',
+            headers: {
+                'content-type': 'application/json',
+            },
+            body: JSON.stringify({
+                username: 'GEN',
+                password: 'secret'
+            })
+        })
+
+        const cookie = loginResponse.headers.get('set-cookie');
+
+        const { sensorSyncLogs, temperatureLogSyncLogs, temperatureBreachSyncLogs } = logs.reduce<Record<string, SyncLog[]>>((acc, log) => {
+            const { type } = log;
+            switch(type) {
+                case ENTITIES.SENSOR:
+                    return { ...acc, sensorSyncLogs: [ ...acc.sensorSyncLogs, log] };
+                case ENTITIES.TEMPERATURE_LOG:
+                    return { ...acc, temperatureLogSyncLogs: [ ...acc.temperatureLogSyncLogs, log] };
+                case ENTITIES.TEMPERATURE_BREACH:
+                    return { ...acc, temperatureBreachSyncLogs: [ ...acc.temperatureBreachSyncLogs, log]}
+                default:
+                    return acc;
+            }
+        }, { sensorSyncLogs: [], temperatureLogSyncLogs: [], temperatureBreachSyncLogs: [] }); 
+        
+        if (cookie) {
+            const pushConfig = {
+                method: 'PUT',
+                headers: {
+                    'content-type': 'application/json',
+                    'cookie': cookie
+                },
+            }
+    
+            await fetch(this.paths.sensor, {
+                ...pushConfig,
+                body: JSON.stringify(sensorSyncLogs.map(log => JSON.parse(log.payload)))
+            })
+
+            await fetch(this.paths.temperatureLog, {
+                ...pushConfig,
+                body: JSON.stringify(temperatureLogSyncLogs.map(log => JSON.parse(log.payload)))
+            })
+
+            await fetch(this.paths.temperatureBreach, {
+                ...pushConfig,
+                body: JSON.stringify(temperatureBreachSyncLogs.map(log => JSON.parse(log.payload)))
```

Comment:
general comment on the impl

1. login to get a cookie (imo this is like a few ms and doesn't matter if we spam it) - if error, stop
2. Query for and paginate all sync outs of a certain type, i.e. `Sensor` if error, stop
2.  With the batch of <=50 records, send them to the specific endpoint. - if error, stop
3. When no batches are left, go to the next entity  i.e. `TemperatureBreach` then `TemperatureLog` and start at (1)

To start doing this sequentially is good, but we could technically do all this with promises in 'parellel' since we don't care about referential integrity